dashboard/datacenter.py

```diff
@@ -37,20 +37,21 @@ async def purge():
         # we do not needs to purge them:
         #   * agents
         #   * nodes
-        #   * hostname_to_ip
-        #   * ip_to_hostname
+        #   * node_id_to_ip
+        #   * node_id_to_hostname
         logger.info("Purge data.")
-        valid_keys = DataSource.ip_to_hostname.keys()
-        for key in DataSource.node_stats.keys() - valid_keys:
+        alive_nodes = set(node_id
+                          for node_id, node_info in DataSource.nodes.items()
+                          if node_info["state"] == "ALIVE")
+        for key in DataSource.node_stats.keys() - alive_nodes:
```

Comment:
I remember you said we'd want to list history of dead nodes before? Isn't this removing that feature completely?

Reply:
I want to reserve a limited count of DEAD nodes, but the `GcsNodeInfo` does not have a timestamp. So, I just removed all DEAD nodes in this PR.
If this is your concern, I can add this feature in this PR.